src/level_four/core/controller/initial_device_provision_contoller.c

```diff
@@ -444,26 +446,29 @@ void lock_all_slots(){
 
     atecc_data.retries = DEFAULT_ATECC_RETRIES;
     bool lock = false;
+    uint32_t err_count=0;
     do
     {
-        if(atecc_data.status != ATCA_SUCCESS){
+        if(atecc_data.status != ATCA_SUCCESS || err_count != 0){
```

Comment:
Here we are relying on previous status value in global variable `atecc_data.status`.
What if the previous status was not `ATCA_SUCCESS`.

Hard reset variable to FAILURE state as done in line #447 (`atecc_data.retries = DEFAULT_ATECC_RETRIES;`) & move this check in the bottom of the do-while loop.